src/traces/parcoords/defaults.js

```diff
@@ -0,0 +1,101 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+var attributes = require('./attributes');
+var hasColorscale = require('../../components/colorscale/has_colorscale');
+var colorscaleDefaults = require('../../components/colorscale/defaults');
+
+var handleLineDefaults = function lineDefaults(traceIn, traceOut, defaultColor, layout, coerce) {
+
+    coerce('line.color', defaultColor);
+    coerce('line.colorscale');
+
+    if(hasColorscale(traceIn, 'line')) {
+        colorscaleDefaults(traceIn, traceOut, layout, coerce, {prefix: 'line.', cLetter: 'c'});
+    }
+    else {
+        coerce('line.color', defaultColor);
+    }
+};
+
+
+function dimensionsDefaults(traceIn, traceOut) {
+    var dimensionsIn = traceIn.dimensions || [],
+        dimensionsOut = traceOut.dimensions = [];
+
+    var dimensionIn, dimensionOut, i;
+    var commonLength = Infinity;
+
+    dimensionsIn.splice(60); // parcoords supports up to this many dimensions
+
+    function coerce(attr, dflt) {
+        return Lib.coerce(dimensionIn, dimensionOut, attributes.dimensions, attr, dflt);
+    }
+
+    for(i = 0; i < dimensionsIn.length; i++) {
+        dimensionIn = dimensionsIn[i];
+        dimensionOut = {};
+
+        if(!Lib.isPlainObject(dimensionIn) || !Array.isArray(dimensionIn.values)) {
```

Comment:
when :arrow_double_up: is addressed, can you add a jasmine default test case for the desired behaviour?